form/src/main/java/org/dhis2/form/data/EventRepository.kt

```diff
@@ -426,21 +427,31 @@ class EventRepository(
                 PeriodSelector(
                     type = periodType,
                     minDate = getPeriodMinDate(periodType),
-                    maxDate = dateUtils.today,
+                    maxDate = getStartOfDay(Date()),
                 )
             } else {
                 null
             }
         }
     }
 
+    private fun getStartOfDay(date: Date): Date {
```

Comment:
I see that you have created this three times, what do you think about create it once in DateUtils and use it from there?